### PR TITLE
fix: disable x filling for line and areas

### LINKED ISSUE #825: Line series with different x values are hidden
**Describe the bug**
Using `LineSeries` with different x values causes line to not be drawn correctly.

**To Reproduce**
1. Go to https://codesandbox.io/s/priceless-wind-clh9f?file=/src/App.tsx
2. Notice the lack of connected lines
3. Toggle overall data, notice correct rendering of browser data
4. Toggle browser data, notice correct rendering of overall data

> Also see https://dev-next-oblt.elastic.dev/app/csm?rangeFrom=now-24h&rangeTo=now&serviceName=elastic-co-frontend

**Expected behaviour**
All lines are connected between respective points

**Screenshots**
image

**Version (please complete the following information):**
 - Elastic Charts: `21.0.0` started form https://github.com/elastic/elastic-charts/pull/751

**Additional context**
The `overall_data` has different x values that the `browser_data` which is likely causing the discontinuity.

### PR DESCRIPTION
**Pull request by markov00**
## Summary

This fix removes the needs of filling the missing x values on line and area charts in the following
case: non-stacked line or area chart, with continuous scale and no fit function.

BREAKING CHANGE: On non-stacked line or area charts, with a continuous x scale and no fit function,
the line between consecutive points, independently from the other data series, will be a continuous
connecting line.

fix #825

*Before the fix*

Screenshot 2020-09-23 at 23 55 10

*After the fix*

Screenshot 2020-09-23 at 23 54 44

### To Do
- [ ] remove the playground demo data after review
- [ ] add a VRT check

### Checklist

Delete any items that are not applicable to this PR.

- [x] Any consumer-facing exports were added to `src/index.ts` (and stories only import from `../src` except for test data & storybook)
- [ ] Proper documentation or storybook story was added for features that require explanation or tutorials
- [ ] Unit tests were updated or added to match the most common scenarios

### CODE REVIEW

**Review by nickofthyme — COMMENTED** [review]
LGTM, path ordering story is still good.

**Comment by nickofthyme** [comment]
It looks like the lastValue was affected to pull last value for each series not necessarily the last x domain value. I think this could be really confusing if they are all from different x values.

image

File: src/chart_types/xy_chart/utils/fill_series.ts [hunk 1]
```diff
@@ -74,3 +97,9 @@ export function fillSeries(
   });
   return filledSeries;
 }
+
+function isXFillNotRequired(spec: BasicSeriesSpec, groupScaleType: ScaleType, isStacked: boolean) {
+  const onlyNoFitAreaLine = (isAreaSeriesSpec(spec) || isLineSeriesSpec(spec)) && !spec.fit;
+  const onlyContinuous = groupScaleType === ScaleType.Linear || groupScaleType === ScaleType.Time;
```
[the remark this answers is not in the document]
Comment (markov00) [reply src/chart_types/xy_chart/utils/fill_series.ts:103]: with `Ordinal` scales it should work as before. the line, usually indicates a continuity on the data, using ordinal aka categorical scale the continuity is not the primary thing.
Let's stick with that for the moment, if there is the case to link discontinued points also on categorical scales we can fix that later, what do you think?

File: integration/tests/line_stories.test.ts [hunk 2]
```diff
@@ -56,4 +56,17 @@ describe('Line series stories', () => {
...
+        'http://localhost:9001/?path=/story/line-chart--discontinuous-data-points&knob-enable fit function=false&knob-switch to area=',
+      );
+    });
+
+    it('no fit function', async () => {
+      await common.expectChartAtUrlToMatchScreenshot(
+        'http://localhost:9001/?path=/story/line-chart--discontinuous-data-points&knob-enable fit function=true&knob-switch to area=',
+      );
+    });
+  });
```
Comment (nickofthyme) [inline integration/tests/line_stories.test.ts:71]: 😍

**Review by nickofthyme — APPROVED** [review]
Code changes LGTM. Verified fixes issue.

File: src/chart_types/xy_chart/utils/fill_series.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (nickofthyme) [reply src/chart_types/xy_chart/utils/fill_series.ts:103]: Yeah that's fine. The fitting function logic does account for ordinal values, I was just wondering if this eliminated that functionality. Looks like it is still working as it was....

https://github.com/elastic/elastic-charts/blob/ae6fe1e773818efed7a0d6a4cfbcd660363b46d4/integration/tests/__image_snapshots__/mixed-stories-test-ts-mixed-series-stories-fitting-functions-area-charts-ordinal-dataset-no-end-value-should-display-correct-fit-for-type-carry-1-snap.png

**Pull request merged**